velox/functions/prestosql/types/BigintEnumType.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BigintEnumType.h"
+#include <boost/algorithm/string.hpp>
+#include <stdexcept>
+
+namespace facebook::velox {
+
+// Type info (enum name and key-value mapping) is passed from the coordinator in
+// the following format as a string:
+// e.g., "test.enum.mood:BigintEnum(test.enum.mood{“CURIOUS”:-2, “HAPPY”:0})".
+// This function checks and extracts the following:
+// 1. Check that the type is correct (BigintEnum).
+// 2. Extract the name of the enum type ("test.enum.mood").
+// 3. Construct a map of the enum mapping ({"CURIOUS": -2, "HAPPY": 0}).
+void BigintEnumType::parseTypeInfo(const std::string& typeInfoString) {
```

Comment:
What's the purpose of this parsing? It doesn't look like we use the results in any way. Can we remove this logic from Velox?

CC: @amitkdutta

Reply:
@HeidiHan0000 I see. I got confused by parse method returning void. I see now that it updates enumMap_. This is an anti-pattern though. A better design is to have this method as static and return the map.

However, can we move parsing logic to the caller and require a map as ctor argument?